api/v1alpha1/types.go

```diff
@@ -86,12 +86,31 @@ type Validation struct {
 	Message    string `json:"message,omitempty"`
 }
 
+// ExternalRef is a reference to an external resource.
+// It allows the user to specify the Kind, Version, Name and Namespace of the resource
+// to be read and used in the Graph.
+type ExternalRef struct {
+	// +kubebuilder:validation:Optional
+	Group string `json:"group,omitempty"`
+	// +kubebuilder:validation:Required
+	Kind string `json:"kind,omitempty"`
+	// +kubebuilder:validation:Required
+	Version string `json:"version,omitempty"`
+	// +kubebuilder:validation:Required
+	Name string `json:"name,omitempty"`
```

Comment:
nit: you do not need omitempty when a field is required (and i might have missed that as well in other fields/structs)